Automattic/wp-calypso
# Move composite-checkout-wpcom into calypso (4)

#### Changes proposed in this Pull Request

The (unpublished) `@automattic/composite-checkout-wpcom` package was built as a way to modularize parts of calypso that were related solely to the composite checkout project, but the cost for this modularization in our monorepo has been too high. This PR moves all of the files in the package into the main calypso package instead.

The largest issue we encountered was that any changes to files in the separate package caused a massive rebuild; a change to a file in calypso might take around 20 seconds, but a change to a file in `packages/` might take 3 minutes. It also uses a tremendous amount of processing power, quickly draining the battery on most laptops.

The second largest issue is that the implementation of composite checkout requires files from other parts of calypso, but to avoid creating circular package dependencies we were forced to pass files through several layers of functions and components in order to get them where they need to be. With the files in the same package, we can simply import them where they are needed.

This PR aims to just move the whole package into calypso for the sake of speed and the smallest amount of changes. Later PRs can refactor to undo hacks that had been added in the package which are no longer needed.

#### To do

- [x] Copy files into calypso
- [x] Change all file references to copied files
- [x] Copy tests into calypso and adapt for new setup
- [x] Remove old package

#### Testing instructions

- Run client tests and make sure they pass: `npx jest -c=test/client/jest.config.js "client/my-sites/checkout/composite-checkout"`.
- Visit composite checkout with a plan in your cart and verify that your cart is displayed correctly.
- Visit composite checkout with a domain in your cart and verify that you see the contact form correctly.
